### ISSUE TITLE
Splatting a tuple with an unpacked variadic prefix unions the appended element type into every parameter slot

### ISSUE THREAD

**Issue by dsfaccini** [astral-sh/ty#4295]
> This issue was posted by Claude Code using claude-fable-5 on behalf of David.

### Summary

Appending an element to a `tuple[Any, ...]` and splatting the result into a call makes ty match `Any | <appended type>` against **every** parameter, so the call is rejected on every slot. The union comes from the two possible positions of the appended element (`tuple[*tuple[Any, ...], Agent]`), and since `Any | Agent` isn't gradual, nothing is assignable to it.

Playground link: PENDING

```py
from typing import Any


class Toolset: ...


class Agent: ...


def temporalize(toolset: Toolset, prefix: str, agent: Agent | None = None) -> None: ...


def build(pass_agent: bool, toolset: Toolset, agent: Agent) -> None:
    args: tuple[Any, ...] = (toolset, 'prefix')
    if pass_agent:
        args = (*args, agent)
    temporalize(*args)
```

```console
uvx ty check repro.py
```

### Actual result

```
repro.py:17:17: error[invalid-argument-type] Argument to function `temporalize` is incorrect: Expected `Toolset`, found `Any | Agent`
repro.py:17:17: error[invalid-argument-type] Argument to function `temporalize` is incorrect: Expected `str`, found `Any | Agent`
Found 2 diagnostics
```

`reveal_type(args)` after the `if` gives `tuple[Any, ...] | tuple[*tuple[Any, ...], Agent]`, which looks right; the diagnostics come from how the second arm is splatted into the call.

### Expected result

No diagnostics. The declared element type of the variadic part is `Any`, so every parameter slot should accept it.

### Notes

- ty 0.0.72 via `uvx ty check`.
- pyright and mypy are both clean on this file.
- Adjacent but not the same as #3649: there the extra positional argument is written at the call site (`foo(*x, y)`) and the diagnostic is `too-many-positional-arguments`; here everything is inside the splatted tuple and the diagnostic is `invalid-argument-type`.
- Real-world source: [`durable_exec/temporal/_agent.py`](https://github.com/pydantic/pydantic-ai/blob/main/pydantic_ai_slim/pydantic_ai/durable_exec/temporal/_agent.py) in pydantic-ai, which builds a positional-argument tuple and conditionally appends one more element depending on the arity of a user-supplied callable, for backwards compatibility.

**Comment by carljm** [comment]
Thanks! Yeah, we should match up the arguments more precisely here. Our inferred type for `args` -- `tuple[Any, ...] | tuple[*tuple[Any, ...], Agent]` -- is correct, and more precise than pyright or mypy infer (just `tuple[Any, ...]`). But if we will infer this more precise type, then we need to handle the call argument unpacking with more precision than we currently do.
